src/witness.rs

```diff
@@ -196,12 +201,15 @@ impl CompiledCircuit {
         // compute each rows' vars, except for the deferred ones (public output)
```

Comment:
so I think what comes after should be part of the backend as well, this is very specific to kimchi/plonk

Reply:
so perhaps the backend should have a `fn generate_witness()` which we could call at this point